### PR TITLE
Swift 4.2.1 conversion, Watson 1.2

### PR DESCRIPTION
**Pull request by christiancompton**
This includes changes to upgrade the Watson SDK to version 1.1.0, including the necessary API changes in the *Custom* project's view controller.

This also upgrades the project to be a `Swift 4.2.1` project, and removes some warning from the story boards and other configuration files in `Xcode 10.1`.

I removed the legacy API Key initializer fallback logic - the Watson SDK dropped support for legacy Visual Recognition credentials in the 1.0.0 release.

### CODE REVIEW

**Comment by christiancompton** [comment]
### Remaining work:
- [ ] Upgrade the Simple project to use the Watson SDK @ 1.2.0
- [ ] Add icons to Simple Project
- [ ] Complete Testing on Hardware for both projects.

**Comment by christiancompton** [comment]
I have completed testing and validated that the code works.  This repository lacks app icons.  I will create a new issue for this, as it will be required as a starter kit.

**Comment by devinaconley** [comment]
Also tested end to end and confirmed updates working

File: Core ML Vision Custom/Cartfile [hunk 1]
```diff
@@ -1 +1 @@
-github "watson-developer-cloud/swift-sdk" == 0.33.0
+github "watson-developer-cloud/swift-sdk" ~> 1.2
```
[the remark this answers is not in the document]
Comment (AnthonyOliveri) [reply Core ML Vision Custom/Cartfile]: Yep. I'll try my best not to break anything :)

**Pull request merged**